### PR TITLE
Introduce swift-xcodegen

### LINKED ISSUE #70604: Contributor Experience: Automate the Xcode workspace generation process
### Description

Having to manually go through [the steps](https://github.com/apple/swift/blob/b01dbc5c9caeb4da2fc4676fa9900e3443ce6a89/docs/HowToGuides/GettingStarted.md#using-ninja-with-xcode) to create a workspace and add stuff to it is... unfortunate. It would be nice to have a separate command line tool that both generates projects via `build-script` and automates at least this part of the setup.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by hamishknight**
This is a tool specifically designed to generate Xcode projects for the Swift repo (as well as a couple of adjacent repos such as LLVM and Clang). It aims to provide a much more user-friendly experience than the CMake Xcode generation (`build-script --xcode`). Forum post to follow.

Resolves #70604

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please smoke test

**Comment by hamishknight** [comment]
@swift-ci please smoke test

**Comment by hamishknight** [comment]
Looks like `swift-xcodegen.test` is failing in CI (#77407). Passes locally, going to disable for now and will resolve tomorrow.

File: utils/swift-xcodegen/README.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  --compiler-libs/--no-compiler-libs
+                          Generate targets for compiler libraries (default: --compiler-libs)
+  --compiler-tools/--no-compiler-tools
+                          Generate targets for compiler tools (default: --compiler-tools)
+  --docs/--no-docs        Add doc groups to the generated projects (default: --docs)
+  --stdlib, --stdlib-cxx/--no-stdlib, --no-stdlib-cxx
+                          Generate a target for C/C++ files in the standard library (default: --stdlib)
+  --stdlib-swift/--no-stdlib-swift
+                          Generate targets for Swift files in the standard library. This requires
+                          using Xcode with with a main development snapshot (and as such is disabled
```
Comment (jamieQ) [inline utils/swift-xcodegen/README.md:77]: nit:
```suggestion
                          using Xcode with a main development snapshot (and as such is disabled
```

**Comment by jamieQ** [comment]
wow, this is very exciting! i expect upon merge, this will also resolve https://github.com/swiftlang/swift/issues/70604?

**Comment by hamishknight** [comment]
🚢

File: utils/swift-xcodegen/README.md (hunk 1 above)
Comment (hamishknight) [reply utils/swift-xcodegen/README.md:77]: Resolving in https://github.com/swiftlang/swift/pull/77415

**Comment by amritpan** [comment]
@hamishknight I'm running into this error when I attempt to generate an Xcode project using the new setup: 

``` **error**: keyNotFound(CodingKeys(stringValue: "output", intValue: nil), Swift.DecodingError.Context(codingPath: [_JSONKey(stringValue: "Index 0", intValue: 0)], debugDescription: "No value associated with key CodingKeys(stringValue: \"output\", intValue: nil) (\"output\").", underlyingError: nil))```

This is after building the toolchain for `--debug-swift`

The full output is: 
```
utils/generate-xcode /Users/amritpankaur/swift-project/build/Ninja-DebugAssert
Building for production...
[1/1] Write swift-version--58304C5D6DBC2206.txt
Build complete! (0.24s)
info: Generating project for '/Users/amritpankaur/swift-project/build/Ninja-DebugAssert'...
error: keyNotFound(CodingKeys(stringValue: "output", intValue: nil), Swift.DecodingError.Context(codingPath: [_JSONKey(stringValue: "Index 0", intValue: 0)], debugDescription: "No value associated with key CodingKeys(stringValue: \"output\", intValue: nil) (\"output\").", underlyingError: nil))
```

**Comment by hamishknight** [comment]
@amritpan should be fixed by #77415

**Pull request merged**